src/main/java/com/google/cloud/tools/appengine/cloudsdk/serialization/CloudSdkComponent.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.appengine.cloudsdk.serialization;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Representation of a CloudSdkComponent. Used for json serialzation/deserialization.
+ */
+public class CloudSdkComponent {
```

Comment:
At the same time, it's not possible to have more than any one component, the components don't change that often, and you can always catch those JsonParseExceptions so they're not thrown.
How does a class allow for more flexibility?